modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFileBootstrap.pm

```diff
@@ -0,0 +1,226 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2020] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 NAME
+
+ Bio::EnsEMBL::Production::Pipeline::Webdatafile::WebdataFileBootstrap;
+
+=head1 DESCRIPTION
+  Compute bootstrap step for webdatafile dumps
+
+=cut
+
+package Bio::EnsEMBL::Production::Pipeline::Webdatafile::WebdataFileBootstrap;
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Registry;
+use base qw/Bio::EnsEMBL::Production::Pipeline::Common::Base/;
+use Path::Tiny qw(path);
+use Array::Utils qw(intersect);
+use Path::Tiny qw(path);
+use Carp qw/croak/;
+use JSON qw/decode_json/;
+use Bio::EnsEMBL::Production::Pipeline::Webdatafile::lib::GenomeLookup;
+sub param_defaults {
+  my ($self) = @_;
+  return {
+    %{$self->SUPER::param_defaults},
+  };
+}
+
+
+sub run {
+
+  my ($self) = @_;
+  my $species = $self->param('species');
+  my $current_step = $self->param('current_step') ;
+  my $output = $self->param('output_path');
+  my $app_path = $self->param('app_path'); 
+  my $genome_data = {
+    dbname     => $self->param('dbname'),
+    gca        => $self->param('gca'),
+    genome_id  => $self->param('genome_id'),
+    species    => $self->param('species'),
+    version    => $self->param('version'),
+    type       => $self->param('division'),  
+    root_path  => path($self->param('root_path'))
+  };
+
+  my $dba = Bio::EnsEMBL::Registry->get_DBAdaptor( $self->param('species'), $self->param('group') );
+  my $lookup = Bio::EnsEMBL::Production::Pipeline::Webdatafile::lib::GenomeLookup->new("genome_data" => $genome_data); 
+  my $genome = $lookup->get_genome('1');
+
+  #bootstrap
+  $self->get_assembly_report($genome, $dba);
+  $self->report_to_chrom_lookups($genome);
+  $self->generate_genome_summary_bed($genome);
+}
+
+sub get_assembly_report {
+
+  my ($self, $genome, $dba) = @_;
+
+  my $target_path = $genome->genome_report_path()->stringify;
+  open(my $fh, ">", $target_path);
+
+  my $statement = $self->prepare_sql($genome->species_id());
```

Comment:
$statement is not used anywhere, should that be passed via `-SQL` in the next line?

Reply:
i will reove this line , as species id is always 1 i passed it  prepare_sql(1) 
 $query_result = $dba->dbc->sql_helper()->execute( -SQL => $self->prepare_sql(1) );

Reply:
`species_id` is only 1 for non-collection databases. For collection dbs it is used to distinguish between the species within a database - I don't understand exactly what all the code is doing, but I'd have thought you would need to account for it taking a range of values.

Reply:
this sql statement is to create report file similar to  NCBI assembly report (https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA_000001405.27_GRCh38.p12/GCA_000001405.27_GRCh38.p12_assembly_report.txt) as chromosome names in assembly reports from NCBI sometimes differed from chromosome names used at EBI. Andy suggested a SQL query for our core database for generating the same data as in NCBI assembly report . we generated genome_report.txt file has the same format as NCBI assembly report.

Reply:
as you mention species id will differ for collection DB , i will consider this and make changes accordingly